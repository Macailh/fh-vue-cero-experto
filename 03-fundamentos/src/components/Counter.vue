<template>
  <h2>{{ customTitle }}</h2>
  <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
  <p data-test-id="counter">{{ counter }}</p>

  <div>
    <button @click="decrease">-1</button>
    <button v-on:click="increase">+1</button>
  </div>
</template>

<script>
export default {
    props: {
        title: String,
        start: {
            type: Number,
            required: true,
            default: 1,
            validator(value) {
                return value >= 0
            }
        }
    },
    name: 'CounterName',
    data() {
        return {
            counter: this.start
        }
    },
    methods: {
        getSquareValue() {
            return this.counter * this.counter
        },
        increase() {
            // this.counter = this.counter + 1
            this.counter++
        },
        decrease() {
            this.counter--
        }
    },
    computed: {
        squareCounter() {
            return this.counter * this.counter
        },
        customTitle() {
            return this.title ? this.title : 'Counter'
        }
    }
}
</script>

<style>
button {
    background-color: darkseagreen;
    border-radius: 5px;
    border: 1px solid white;
    color: white;
    cursor: pointer;
    margin: 0 5px;
    padding: 5px 15px;
    transition: 0.25s ease-in-out;
}

button:hover {
    background-color: #5aa67b;
    transition: 0.25s ease-in-out;
}
</style>