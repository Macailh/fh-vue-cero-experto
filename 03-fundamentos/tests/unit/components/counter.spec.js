import { shallowMount } from "@vue/test-utils"
import Counter from "@/components/Counter.vue"


describe('Counter Component', () => {
    test('match with snapshot', () => {
        const wrapper = shallowMount(Counter)
        expect(wrapper.html()).toMatchSnapshot()
    })
})