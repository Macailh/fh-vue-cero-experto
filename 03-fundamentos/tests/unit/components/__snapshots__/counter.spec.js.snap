// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component match with snapshot 1`] = `
<h2>Counter</h2>
<p>1 <sup>2</sup> = 1</p>
<div><button>-1</button><button>+1</button></div>
`;
